### PR TITLE
[WS-Passive-Scan] Email disclosure scan with javascript 

### PR DESCRIPTION
**Pull request by ManosMagnus**
This script scans incoming WebSocket messages for email address.

_____________________________________________________
TODOs:

- [x] Add help content 
- [x] Based on  #2089

### CODE REVIEW

**Comment by ManosMagnus** [comment]
Ready for review

File: addOns/websocket/src/main/javahelp/org/zaproxy/zap/extension/websocket/resources/help/contents/pscanrules.html [hunk 1]
```diff
@@ -11,5 +11,18 @@ <H1>WebSocket Passive Scan Rules</H1>
         <h2 id="scripts">Scripts</h2>
         Scripts which are included by default in the add-on and they implement the following WebSocket passive scan rules:
 
+        <h3>Information Disclosure: Email address</h3>
+
+        This script scans incoming WebSocket messages for email address. Email address may be not specifically meant for end user consumption.<br>
```
Comment (kingthorin) [inline addOns/websocket/src/main/javahelp/org/zaproxy/zap/extension/websocket/resources/help/contents/pscanrules.html]: "address" > "addresses" (or "address(es)")

File: addOns/websocket/src/main/zapHomeFiles/scripts/templates/websocketpassive/Email Disclosure.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+// This script scans incoming WebSocket messages for email address.
+
+// * Author: Manos Kirtas (Manos Kirtas)
+// * Based on: community-scripts/passive/Find Emails.js
+
+OPCODE_TEXT = 0x1;
+RISK_INFO = 0;
+CONFIDENCE_HIGH = 3;
+
+var emailRegex = new RegExp("([a-zA-Z0-9_.+-]+@[a-zA-Z0-9]+[a-zA-Z0-9-]*\.[a-zA-Z0-9-.]*[a-zA-Z0-9]{2,})", 'gmi');
```
Comment (kingthorin) [inline addOns/websocket/src/main/zapHomeFiles/scripts/templates/websocketpassive/Email Disclosure.js]: If it's flagged `gmi` (`i` specifically) do you need to specify both `a-zA-Z`?

File: addOns/websocket/src/main/zapHomeFiles/scripts/templates/websocketpassive/Email Disclosure.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    }
+    var message = String(msg.getReadablePayload()).valueOf();
+    var matches;
+
+    if((matches = message.match(emailRegex)) != null) {
+        matches.forEach(function(evidence){
+            helper.newAlert()
+                .setRiskConfidence(RISK_INFO, CONFIDENCE_HIGH)
+                .setName("Email address found in WebSocket message (script).")
+                .setDescription("Email address were found at WebSocket Message.")
```
Comment (kingthorin) [inline addOns/websocket/src/main/zapHomeFiles/scripts/templates/websocketpassive/Email Disclosure.js]: Were implies plural so this should use either "addresses" or "address(es)"

File: addOns/websocket/src/main/zapHomeFiles/scripts/templates/websocketpassive/Email Disclosure.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+                .setSolution("Remove emails that are not public")
```
Comment (kingthorin) [inline addOns/websocket/src/main/zapHomeFiles/scripts/templates/websocketpassive/Email Disclosure.js]: Period

File: addOns/websocket/src/main/zapHomeFiles/scripts/templates/websocketpassive/Email Disclosure.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+CONFIDENCE_HIGH = 3;
+
+var emailRegex = new RegExp("([a-zA-Z0-9_.+-]+@[a-zA-Z0-9]+[a-zA-Z0-9-]*\.[a-zA-Z0-9-.]*[a-zA-Z0-9]{2,})", 'gmi');
+
+function scan(helper,msg) {
+
+    if(msg.opcode != OPCODE_TEXT || msg.isOutgoing){
+        return;
+    }
+    var message = String(msg.getReadablePayload()).valueOf();
```
Comment (thc202) [inline addOns/websocket/src/main/zapHomeFiles/scripts/templates/websocketpassive/Email Disclosure.js]: Remove `valueOf()`?

File: addOns/websocket/src/main/javahelp/org/zaproxy/zap/extension/websocket/resources/help/contents/pscanrules.html [hunk 6]
```diff
@@ -11,5 +11,18 @@ <H1>WebSocket Passive Scan Rules</H1>
...
+
+        <br>
+        <table border="1"  width = "400">
+            <caption>Default Values</caption>
+            <tr><th>Risk</th><td>Info</td></tr>
+            <tr><th>Confidence</th><td>Info</td></tr>
+            <tr><th>CWE ID</th><td>200: Information Exposure</td></tr>
+            <tr><th>WASC ID</th><td>12: Information Leakage</td></tr>
```
Comment (kingthorin) [inline addOns/websocket/src/main/javahelp/org/zaproxy/zap/extension/websocket/resources/help/contents/pscanrules.html]: I think this should be 13 like the alert for Info Leakage.

File: addOns/websocket/src/main/zapHomeFiles/scripts/templates/websocketpassive/Email Disclosure.js (hunk 3 above)
Comment (thc202) [reply addOns/websocket/src/main/zapHomeFiles/scripts/templates/websocketpassive/Email Disclosure.js]: Probably better to change to "was", the alert contains just one.

Comment (kingthorin) [reply addOns/websocket/src/main/zapHomeFiles/scripts/templates/websocketpassive/Email Disclosure.js]: Good call, same for the help content.

File: addOns/websocket/src/main/javahelp/org/zaproxy/zap/extension/websocket/resources/help/contents/pscanrules.html (hunk 1 above)
Comment (ManosMagnus) [reply addOns/websocket/src/main/javahelp/org/zaproxy/zap/extension/websocket/resources/help/contents/pscanrules.html]: > Good call, same for the help content.
So we keep this as it is?

Comment (kingthorin) [reply addOns/websocket/src/main/javahelp/org/zaproxy/zap/extension/websocket/resources/help/contents/pscanrules.html]: Hmmm it does read a bit awkwardly. Ummm.... 

How about:
"This script scans incoming WebSocket messages for email addresses. Email addresses may be not specifically meant for end user consumption."

File: addOns/websocket/src/main/zapHomeFiles/scripts/templates/websocketpassive/Email Disclosure.js (hunk 3 above)
Comment (kingthorin) [reply addOns/websocket/src/main/zapHomeFiles/scripts/templates/websocketpassive/Email Disclosure.js]: "An email address was found in a WebSocket Message."
(Or maybe "this", though the alert is associated with the upgrade http message right, not the actual WS message.....)

**Comment by ManosMagnus** [comment]
Rebased and comments are resolved

File: addOns/websocket/src/main/javahelp/org/zaproxy/zap/extension/websocket/resources/help/contents/pscanrules.html (hunk 1 above)
Comment (ManosMagnus) [reply addOns/websocket/src/main/javahelp/org/zaproxy/zap/extension/websocket/resources/help/contents/pscanrules.html]: Fixed in 944da55ca

File: addOns/websocket/src/main/zapHomeFiles/scripts/templates/websocketpassive/Email Disclosure.js (hunk 3 above)
Comment (ManosMagnus) [reply addOns/websocket/src/main/zapHomeFiles/scripts/templates/websocketpassive/Email Disclosure.js]: Fixed in 944da55ca

File: addOns/websocket/src/main/zapHomeFiles/scripts/templates/websocketpassive/Email Disclosure.js (hunk 5 above)
Comment (ManosMagnus) [reply addOns/websocket/src/main/zapHomeFiles/scripts/templates/websocketpassive/Email Disclosure.js]: Fixed in 450043c95

File: addOns/websocket/src/main/zapHomeFiles/scripts/templates/websocketpassive/Email Disclosure.js (hunk 2 above)
Comment (ManosMagnus) [reply addOns/websocket/src/main/zapHomeFiles/scripts/templates/websocketpassive/Email Disclosure.js]: Fixed in 450043c95

**Comment by thc202** [comment]
Thank you.

**Comment by ManosMagnus** [comment]
Thank you!

Commits are squashed

**Pull request merged**